activesupport/lib/active_support/lazy_load_hooks.rb

```diff
@@ -70,7 +74,25 @@ def on_load(name, options = {}, &block)
     #
     # In the case of the above example, it will execute all hooks registered
     # for +:active_record+ within the class +ActiveRecord::Base+.
-    def run_load_hooks(name, base = Object)
+    def run_load_hooks(name, base = Object, after: nil)
+      if ActiveSupport::LazyLoadHooks.warn_on_premature_loading && after && @loaded[after].blank?
+        puts <<~MSG
+
+          Load hook #{name.inspect} was called before load hook #{after.inspect}.
+          Prematurely loading frameworks may slow down your boot time and could
+          cause conflicts with load order and boot of your application.
+
+          Consider wrapping your code with an on_load hook:
+
+              ActiveSupport.on_load(#{name.inspect}) do
+                # your code
+              end
+
+        MSG
+        if ::Rails.respond_to?(:backtrace_cleaner)
```

Comment:
Active Support should not know about `Rails` or the existence of the application. Maybe the backtrace cleaner should just be skipped? I don't see a problem with showing everyting, because if people are cleaning the backtrace of gems, they will not see who is trying to load the framework too early.

Reply:
It's showing the whole backtrace now.